### PR TITLE
Fix missing OwnerReference when updating dapr service

### PR DESCRIPTION
**Pull request by sunzhaochang**
# Description

Add OwnerReference when updating dapr service.

## Issue reference

Please reference the issue this PR will close:  #4849 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by daixiang0** [comment]
Could we add a test for it to ensure fix?

**Comment by sunzhaochang** [comment]
The test has been added. PTAL  @daixiang0

File: pkg/operator/handlers/dapr_handler_test.go [hunk 1]
```diff
@@ -129,6 +131,48 @@ func TestCreateDaprServiceAppIDAndMetricsSettings(t *testing.T) {
...
+	err = testDaprHandler.patchDaprService(ctx, myDaprService, deployment, actualService)
+	assert.NoError(t, err)
+	err = cli.Get(ctx, myDaprService, &actualService)
+	assert.NoError(t, err)
+	assert.Equal(t, "test", actualService.ObjectMeta.Annotations[appIDAnnotationKey])
+	assert.Equal(t, "true", actualService.ObjectMeta.Annotations["prometheus.io/scrape"])
+	assert.Equal(t, "/", actualService.ObjectMeta.Annotations["prometheus.io/path"])
+	assert.Len(t, actualService.OwnerReferences, 1)
+	assert.Equal(t, "Deployment", actualService.OwnerReferences[0].Kind)
+	assert.Equal(t, "app", actualService.OwnerReferences[0].Name)
```
Comment (daixiang0) [inline pkg/operator/handlers/dapr_handler_test.go:173]: Please add test for delete case, ensure service is deleted as well.

Comment (sunzhaochang) [reply pkg/operator/handlers/dapr_handler_test.go:173]: The service with OwnerReference field will be deleted by kubernetes garbage collector in controller manager, but there is no real k8s cluster introduced in the current unit test, we just create the service with fake client. So is there any suggestions for the deleting case?

Comment (daixiang0) [reply pkg/operator/handlers/dapr_handler_test.go:173]: We have e2e tests, you can add it here.

**Comment by yaron2** [comment]
> /ok-to-test

@daixiang0 you need to open a PR to add yourself to the allowed list for oktotest

**Comment by daixiang0** [comment]
> > /ok-to-test
> 
> @daixiang0 you need to open a PR to add yourself to the allowed list for oktotest

Thanks for tip :)

**Comment by yaron2** [comment]
> > > /ok-to-test
> > 
> > 
> > @daixiang0 you need to open a PR to add yourself to the allowed list for oktotest
> 
> Thanks for tip :)
> 
> Seems I am here:
> 
> https://github.com/dapr/dapr/blob/9b07bc23e7321b609b44fe31c554d25a2a4cdcd0/.github/workflows/dapr-bot.yml#L18-L33

Yeah, I wonder what's up with Dapr-bot. Should check logs

**Comment by yaron2** [comment]
@sunzhaochang see failing builds

**Comment by sunzhaochang** [comment]
@yaron2  @daixiang0  I have fixed the failed test, please trigger the test again.

**Review by yaron2 — APPROVED** [review]

**Pull request merged**